### PR TITLE
WCWEB-203

### PR DESCRIPTION
**Pull request by AndriiBalaban**
Updated flow for Product page call to action  buttons when cart is populated

### CODE REVIEW

**Review by aryehischechter — APPROVED** [review]

File: src/resources/js/paymentForm.js [hunk 1]
```diff
@@ -156,26 +156,43 @@ function initPaypalCheckout() {
...
-                        // If single line item, qty 1 and same as button, no need to update
-                        // otherwise, go to cart page
-                        if (response.cart.lineItems.length === 0) {
-                            // set selected item to cart
-                            return updateCartSingleItem();
-                        } else if (response.cart.lineItems.length === 1 && response.cart.lineItems[0].qty === 1) {
-                            if (response.cart.lineItems[0].purchasableId === purchasableId) {
-                                // Cart has selected item already, we can proceed
+                        function cartPageLogic() {
+                            if(response.cart.totalQty > 0) {
```
Comment (eranpolo) [inline src/resources/js/paymentForm.js:170]: I believe the "contract" here is that once we're called, we should call either actions.resolve() or actions.reject(). This "if" might introduce a case where no action is reported. Perhaps the check for totalQty>0 should be added to the logic selection if? (line 189)

**Review by eranpolo — CHANGES_REQUESTED** [review]

File: src/resources/js/paymentForm.js (hunk 1 above)
Comment (AndriiBalaban) [reply src/resources/js/paymentForm.js:170]: @eranpolo please take a look at my latest version

**Review by eranpolo — APPROVED** [review]

File: src/resources/js/paymentForm.js (hunk 1 above)
Comment (eranpolo) [reply src/resources/js/paymentForm.js:170]: cool, approved.

File: src/resources/js/paymentForm.js [hunk 2]
```diff
@@ -157,11 +157,18 @@ function initPaypalCheckout() {
...
+                        // If in cart page action and empty cart, reject
                         // If no line items, add selected item to cart
                         // If single line item, qty 1 and different than button, replace it
                         // If single line item, qty 1 and same as button, no need to update
                         // otherwise, go to cart page
-                        if (response.cart.lineItems.length === 0) {
+                        if(window.location.pathname.startsWith('/cart') && response.cart.totalQty > 0) {
+                            return actions.resolve();
+                        }
+                        else if (window.location.pathname.startsWith('/cart') && response.cart.totalQty == 0) {
```
Comment (aryehischechter) [inline src/resources/js/paymentForm.js]: 3 =

Comment (aryehischechter) [reply src/resources/js/paymentForm.js]: it should be 3 equals. and the pathname check can be a single if and check the totalQty inside of it. so all the cart logic is within one if.

**Pull request merged**